### PR TITLE
Introduce Intl.ListFormat

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change

Attempt to introduce the Intl.ListFormat and polyfill into the codebase. This progresses #14589 further. As an example, the "zone" trigger and condition has been adjusted, but others have been left out to make this as small as possible. 
This should help at translating the descriptions of triggers, conditions and actions in the future. For now, it's hardcoded as english. 

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: #14589
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
   "compilerOptions": {
     // Language Options
-    "target": "ES2017",
-    "lib": [
-      "ES2017",
-      "DOM",
-      "DOM.Iterable",
-      "WebWorker"
-    ],
+    "target": "ES2021",
```
Comment (bramkragten) [inline tsconfig.json:4]: Why this change?

Comment (silamon) [reply tsconfig.json:4]: Intl.ListFormat isn't included in previous targets, es2021 is the first to include it. If there are other possibilities, let me know, since I'm not entirely familiar with how these targets work.

Comment (steverep) [reply tsconfig.json:4]: Even though it might be questionably related to this PR, I do agree the values for `target` and `lib` need to be kept at the latest release or "ESNext".  We don't use TS to transpile and we type check the code we write which is certainly beyond ES 2017.

I did try to set it to either 2022 or next once and got a bunch of valid type errors.  If 2021 passes checks though, we should keep it IMO.

File: src/data/automation_i18n.ts
Comment (steverep) [inline src/data/automation_i18n.ts:1]: Need to add an import for `intl-polyfill` here

File: src/data/automation_i18n.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import {
 } from "../common/entity/compute_attribute_display";
 import { computeStateDisplay } from "../common/entity/compute_state_display";
 import { EntityRegistryEntry } from "./entity_registry";
+import "../../resources/intl-polyfill";
```
Comment (steverep) [inline src/data/automation_i18n.ts:24]:
```suggestion
import "../resources/intl-polyfill";
```

**Review by steverep — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]
